packages/core/src/babel-plugin-card-component-meta.ts

```diff
@@ -0,0 +1,148 @@
+  }
+}
+
+export function babelPluginComponentMeta(babel: typeof Babel) {
+  return {
+    visitor: {
+      Program: {
+        enter(path: NodePath<t.Program>, state: State) {
+          state.importUtil = new ImportUtil(babel.types, path);
+        },
+        exit(path: NodePath<t.Program>, state: State) {
+          path.node.body.push(exportSerializerMap(path, state, babel));
+          path.node.body.push(exportComputedFields(state, babel));
+          path.node.body.push(exportUsedFields(state, babel));
+        },
+      },
+    },
+  };
+}
+
+function exportSerializerMap(path: NodePath<t.Program>, state: State, babel: typeof Babel): t.Statement {
+  let t = babel.types;
+  let { fields, usedFields } = state.opts;
+
+  let map: SerializerModuleMap = {};
+
+  for (const fieldPath of usedFields) {
+    let field = getFieldForPath(fields, fieldPath);
+
+    if (!field) {
+      continue;
+    }
+
+    addFieldToSerializerMap(map, field, fieldPath, path, state);
+  }
+
+  return babel.template(`export const serializerMap = %%serializerMap%%;`)({
+    serializerMap: t.objectExpression(buildSerializerMapProp(map, t)),
+  }) as t.Statement;
+}
+
+function exportComputedFields(state: State, babel: typeof Babel): t.Statement {
+  let t = babel.types;
+
+  return babel.template(`export const computedFields = %%computedFields%%;`)({
+    computedFields: t.arrayExpression(
+      Object.values(state.opts.fields)
+        .filter((field) => field.computed)
+        .map((field) => t.stringLiteral(field.name))
+    ),
+  }) as t.Statement;
+}
+
+function exportUsedFields(state: State, babel: typeof Babel): t.Statement {
+  let t = babel.types;
+
+  return babel.template(`export const usedFields = %%usedFields%%;`)({
+    usedFields: t.arrayExpression(state.opts.usedFields.map((field) => t.stringLiteral(field))),
+  }) as t.Statement;
+}
+
+// TODO: Where
+type SerializerModuleMap = Record<string, t.Identifier>;
+
+function buildSerializerMapProp(
+  serializerMap: SerializerModuleMap,
+  t: typeof Babel.types
+): t.ObjectExpression['properties'] {
+  let props: t.ObjectExpression['properties'] = [];
+
+  for (let fieldPath in serializerMap) {
+    let modulePath = serializerMap[fieldPath];
+    if (!modulePath) {
+      continue;
+    }
+
+    props.push(t.objectProperty(t.stringLiteral(fieldPath), modulePath));
+  }
+
+  return props;
+}
+
+function addFieldToSerializerMap(
+  map: SerializerModuleMap,
+  field: Field,
+  usedPath: string,
+  path: NodePath<t.Program>,
+  state: State
+): void {
+  if (Object.keys(field.card.fields).length) {
+    let { fields } = field.card;
+    for (const name in fields) {
+      addFieldToSerializerMap(map, fields[name], `${usedPath}.${name}`, path, state);
+    }
+  } else {
+    if (!field.card.serializerModule) {
+      return;
+    }
+    // TEMP: Would be nice to have the Proper CardID on the compiled card
+    let cardId = field.card.url.split('/')[field.card.url.split('/').length - 1];
```

Comment:
This is gross, but with the info we have here this is the best we can do. Basically, we have a Date fields under the key birthdate, I want to add an import for it's serializer with the name "DateSerializer". The `Field` definition and the compiled card have know easy way to say that it's a date field. An ideal solution would be to make a similar change that we made to RawCard and have a dedicated ID field.

Reply:
i don't know if its a same assumption to use the last path segment as the ID. the realm itself is responsible for its own ID's and it might decide to create an ID with a "/" in it. but yeah, a dedicated "realm ID" field is probably not a bad idea